packages/slice-machine/components/Simulator/index.tsx

```diff
@@ -58,22 +67,50 @@ export default function Simulator() {
     return <div />;
   }
 
-  const sliceView = useMemo(
-    () => [
-      {
-        sliceID: component.model.id,
-        variationID: variation.id,
+  const sharedSlice = useMemo(
+    () => Slices.fromSM(component.model),
+    [component.model]
+  );
+
+  const initialContent = useMemo<SharedSliceContent>(
+    () =>
+      component.mock?.[0] ||
+      (defaultSharedSliceContent(variation.id) as SharedSliceContent),
+    [component.mock, variation.id]
+  );
+
+  const [editorContent, setContent] = useState(initialContent);
+  const initialApiContent = useMemo(
+    () =>
+      renderSliceMock(sharedSlice, editorContent) as {
```

Comment:
Can you add a quick comment to explain why we cast here, it's gonna be useful for the next guy